### PR TITLE
feat: add writers for USDA, USDC, and USDZ formats

### PR DESCRIPTION
**Pull request by bresilla**
Adds writers for `.usda`, `.usdc`, and `.usdz` — symmetric with the existing readers.

The six commits are structured so they can be cherry-picked independently if you'd rather split this into smaller PRs (field-ordering prerequisite → `sdf::Data` → USDA writer → USDC writer → USDZ writer → ergonomics).

## Tests

- 10 AOUSD compliance + 25 local `.usda` fixtures round-trip through the text writer
- 45 `.usdc` fixtures round-trip through the binary writer
- 8 cross-format `.usda → .usdc → parse` tests
- 5 USDZ archive tests (STORED, 64-byte aligned, read-back via the existing `Archive`)
- 712 total, clippy clean, fmt clean, no new runtime deps

## Known skips (all reader-side gaps)

- Nested dictionaries (`gen_dict`, `ball.maya`, `fender_stratocaster`, `toy_biplane_idle`, `dictionaries.usda`)
- `asset[]` / `pathExpression[]` arrays, `uint[]` arrays
- Splines (materialise as `Value::ValueVec`; USDC has no heterogeneous-array type)

Each skip has a comment explaining why. Happy to take these on as follow-ups if helpful.

### CODE REVIEW

**Comment by bresilla** [comment]
I edited `ROADMAP.md` and marked those changes for 0.4.0 release. Maybe I jumped the gun, that should have been up to you to decide... you can edit them of course :) :)

File: src/sdf/mod.rs [hunk 1]
```diff
@@ -220,26 +228,55 @@ pub type LayerData = Box<dyn AbstractData>;
...
+            slot.1 = value;
+        } else {
+            self.fields.push((key.to_owned(), value));
+        }
+    }
+
+    /// Look up a field by name.
+    pub fn get(&self, key: &str) -> Option<&Value> {
+        self.fields.iter().find(|(k, _)| k == key).map(|(_, v)| v)
+    }
```
[the remark this answers is not in the document]
Comment (mxpv) [reply src/sdf/mod.rs:263]: The performance concern is hypothetical; C++'s SdfData linear-scans the same way.

**Comment by mxpv** [comment]
This is awesome! Thanks for opening.

I pushed a few changes on top:
- Enabled int array compression
- Nested dictionaries
- Fixed a few bugs flagged by copilot
- Increased tests coverage for corner cases.

**Comment by bresilla** [comment]
> This is awesome! Thanks for opening.

niiiice, amazing :)

well i am trying to make this in OpenUSD (its just bevy objects now), but i will make a crate called `bevy-openusd` and then slowly try to make full scenes of Bevy in OpeUSD...

image
